crates/ruff_server/src/server.rs

```diff
@@ -60,6 +61,42 @@ impl Server {
 
         conn.initialize_finish(id, initialize_data)?;
 
+        // Register capabilities
+        conn.sender
+            .send(lsp_server::Message::Request(lsp_server::Request {
+                id: 1.into(),
+                method: "client/registerCapability".into(),
+                params: serde_json::to_value(lsp_types::RegistrationParams {
+                    registrations: vec![lsp_types::Registration {
+                        id: "ruff-server-watch".into(),
+                        method: "workspace/didChangeWatchedFiles".into(),
+                        register_options: Some(serde_json::to_value(
+                            DidChangeWatchedFilesRegistrationOptions {
+                                watchers: vec![
+                                    FileSystemWatcher {
+                                        glob_pattern: types::GlobPattern::String(
+                                            "**/.?ruff.toml".into(),
+                                        ),
+                                        kind: None,
+                                    },
+                                    FileSystemWatcher {
+                                        glob_pattern: types::GlobPattern::String(
+                                            "**/pyproject.toml".into(),
+                                        ),
+                                        kind: None,
+                                    },
+                                ],
+                            },
+                        )?),
+                    }],
+                })?,
+            }))?;
+
+        // Flush response from server (to avoid an unexpected response appearing in the event loop)
```

Comment:
Do you mean response from `client`?
```suggestion
        // Flush response from the client (to avoid an unexpected response appearing in the event loop)
```